src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegion.java

```diff
@@ -304,9 +307,22 @@ public void addAll(final Collection<GATKRead> readsToAdd){
         Utils.validateArg( padding >= 0, () -> "padding must be a positive integer but got " + padding);
         Utils.validateArg( genomeLoc.size() > 0, () -> "GenomeLoc must have size > 0 but got " + genomeLoc);
 
-        return referenceReader.getSubsequenceAt( genomeLoc.getContig(),
+        final String contig = genomeLoc.getContig();
+        final SAMSequenceRecord sequence = Utils.nonNull(referenceReader.getSequenceDictionary().getSequence(contig),
+                () -> "Contig " + contig + " not found in reference. " +
+                        "\nPlease check that you are using a compatible reference for your data." +
+                        "\nReference Sequences: " + listContigsAsString(referenceReader.getSequenceDictionary()));
+
+        return referenceReader.getSubsequenceAt(contig,
```

Comment:
`getSubsequenceAt()` will already throw a `UserException.MissingContigInSequenceDictionary` exception internally if the contig is missing -- is there some way we could rely on that existing check? Ideally we should avoid adding lots of additional sequence dictionary lookups to frequently-called methods if we can possibly help it...

Reply:
It's not clear to me how to avoid it because we need the sequence length from the sequence to make the call to getSubsequenceAt. This pr isn't adding a new lookup, it's just moving it.  We could add a new sequenceDictionary method that automatically clips to the ends to avoid this extra lookup, but that seems like a lot of effort.

Reply:
Yes, you're right, it can't be avoided. This is fine then.